### PR TITLE
8339347: keytool -importpass insists prompting the user even if there is no terminal

### PR DESCRIPTION
**Pull request by haimaychao**
Given the changes to the System.console() behavior, where it no longer returns null when a write pipe is connected to the Java process, keytool needs to be updated to determine whether an interactive terminal is attached.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Warning
&nbsp;⚠️ Found leading lowercase letter in issue title for `8339347: keytool -importpass insists prompting the user even if there is no terminal`

### Issue
 * [JDK-8339347](https://bugs.openjdk.org/browse/JDK-8339347): keytool -importpass insists prompting the user even if there is no terminal (**Bug** - P4)

### Reviewers
 * [Weijun Wang](https://openjdk.org/census#weijun) (@wangweij - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/20855/head:pull/20855` \
`$ git checkout pull/20855`

Update a local copy of the PR: \
`$ git checkout pull/20855` \
`$ git pull https://git.openjdk.org/jdk.git pull/20855/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 20855`

View PR using the GUI difftool: \
`$ git pr show -t 20855`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/20855.diff">https://git.openjdk.org/jdk/pull/20855.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/20855#issuecomment-2329488264)

### CODE REVIEW

**Comment by wangweij** [comment]
Do you need to check `Password::readPassword(InputStream in, boolean isEchoOn)` also? I'm not sure.

**Comment by haimaychao** [comment]
The `Password::readPassword(InputStream in, boolean isEchoOn)` is used to control whether characters are echoed on the screen as the user types the password. In this `keytool` use case, `Password::readPassword(InputStream in)` works as expected (i.e. reading passwords from stdin without any need to control echo), and I'd think additional echo handling seems not relevant here.

File: test/jdk/sun/security/tools/keytool/TestImportPass.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ *          the non-terminal console
+ * @library /test/lib
+ * @run main TestImportPass
+ */
+
+import jdk.test.lib.SecurityTools;
+
+public class TestImportPass {
+    public static void main(String[] args) throws Throwable {
+        SecurityTools.setResponse("pass123", "pass123");
```
[the remark this answers is not in the document]
Comment (haimaychao) [reply test/jdk/sun/security/tools/keytool/TestImportPass.java]: Yes, removed it.

**Review by wangweij — APPROVED** [review]
Looks good to me. Thanks.

**Comment by haimaychao** [comment]
@wangweij Thanks for the review.

**Pull request closed without merging**